cmd/okctl/createapplication.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/oslokommune/okctl/pkg/config/state"
+
+	"github.com/oslokommune/okctl/pkg/okctlapplication"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/oslokommune/kaex/pkg/api"
+	"github.com/oslokommune/okctl/pkg/okctl"
+	"github.com/spf13/cobra"
+)
+
+// CreateApplicationOpts contains all the possible options for "create application"
+type CreateApplicationOpts struct {
+	fullExample bool
+	env         string
+}
+
+// Validate the options for "create application"
+func (o *CreateApplicationOpts) Validate() error {
+	return validation.ValidateStruct(o,
+		validation.Field(&o.fullExample),
+		validation.Field(&o.env),
+	)
+}
+
+// nolint funlen
+func buildCreateApplicationCommand(o *okctl.Okctl) *cobra.Command {
+	opts := &CreateApplicationOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "application",
+		Short: "Create an application template",
+		Long:  "Scaffolds an application.yaml template which can be used to produce necessary Kubernetes and ArgoCD resources",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			var buffer bytes.Buffer
+
+			if opts.fullExample {
```

Comment:
This should be passed in as an argument to `scaffold.NewApplication(...)` or something to this effect and then handled by the internals of that implementation